### PR TITLE
Adding configuration for paygops worflow

### PR DESCRIPTION
**Pull request by ankita10r**
Added workflow configurations for paygops workflow.

Resolved the comments

@avikganguly01 : Please review

### CODE REVIEW

**Review by avikganguly01 — CHANGES_REQUESTED** [review]

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -36,6 +36,11 @@ bpmn:
     international-remittance-payee: "international-remittance-payee-process-{dfspid}"
     international-remittance-payer: "international-remittance-payer-process-{dfspid}"
     mpesa-flow: "mpesa_flow-{dfspid}"
+    mpesa-flow-paygops: "mpesa_flow_paygops-{dfspid}"
+
+
+identifier:
+  secondary: "ACCOUNTID"
```
[the remark this answers is not in the document]
Comment (ankita10r) [reply src/main/resources/application.yml]: Changed it to AMS

File: src/main/java/org/mifos/connector/channel/camel/utils/AMSUtils.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+public class AMSUtils {
+
+    public AMSUtils(){
+
+    }
+    public static final String MSISDN = "MSISDN";
+    public static final String ACCOUNTID = "ACCOUNTID";
+    public static final String FOUNDATIONALID = "FOUNDATIONALID";
+    public static final String AMS_ROSTER = "roster";
```
Comment (avikganguly01) [inline src/main/java/org/mifos/connector/channel/camel/utils/AMSUtils.java:11]: These should not be hardcoded.

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -35,7 +35,11 @@ bpmn:
     gsma-link-based-payment: "gsma_link_transfer"
     international-remittance-payee: "international-remittance-payee-process-{dfspid}"
     international-remittance-payer: "international-remittance-payer-process-{dfspid}"
-    mpesa-flow: "mpesa_flow-{dfspid}"
+    mpesa-flow: "mpesa_flow-{ams}-{dfspid}"
+
+
+ams:
+  name: "roster"
```
Comment (avikganguly01) [inline src/main/resources/application.yml:42]: The ams names should be mapped to the identifier type or default key.

**Review by avikganguly01 — CHANGES_REQUESTED** [review]

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -35,7 +35,25 @@ bpmn:
...
+    -
+      identifier: "foundationalid"
+      value: "paygops"
+    -
+      identifier: "default"
+      value : "paygops"
+
+
+identification:
+  secondary: "accountid"
```
Comment (avikganguly01) [inline src/main/resources/application.yml:55]: What is this configuration for?

File: src/main/java/org/mifos/connector/channel/camel/utils/AMSUtils.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+       ACCOUNTID = ams.get(0).getIdentifier();
+       FOUNDATIONALID = ams.get(1).getIdentifier();
+        AMS_DEFAULT =ams.get(1).getValue();
+        AMS_SECONDARY = ams.get(0).getValue();
+    }
+
+    public static final String MSISDN = "MSISDN";
+    public  String ACCOUNTID;
+    public  String  FOUNDATIONALID;
+    public  String AMS_SECONDARY;
```
Comment (avikganguly01) [inline src/main/java/org/mifos/connector/channel/camel/utils/AMSUtils.java:42]: We discussed that secondary/primary doesn't make sense.

File: src/main/java/org/mifos/connector/channel/camel/utils/AMSUtils.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    public AMSUtils(){
+    }
+
+    @PostConstruct
+    public void postConstruct(){
+       ams  =  amsProps.getGroups();
+       ACCOUNTID = ams.get(0).getIdentifier();
+       FOUNDATIONALID = ams.get(1).getIdentifier();
+        AMS_DEFAULT =ams.get(1).getValue();
+        AMS_SECONDARY = ams.get(0).getValue();
```
Comment (avikganguly01) [inline src/main/java/org/mifos/connector/channel/camel/utils/AMSUtils.java:36]: If you're again hardcoding everything here, then the configuration doesn't make any sense.

File: src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java [hunk 7]
```diff
@@ -302,19 +318,33 @@ private void collectionRoutes(){
...
+                        //default paygops
+                        accountId = "0";
+                        tenantSpecificBpmn = mpesaFlow.replace("{dfspid}", tenantId).replace("{ams}",amsUtils.AMS_DEFAULT);
+
+                    }
+
                     String amount = body.getJSONObject("amount").getString("amount");
 
-                    extraVariables.put("accountId", accountId);
+                    extraVariables.put("accountId", accountId);//will have memoid in case of paygops
```
Comment (avikganguly01) [inline src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java:340]: No hardcoded logic around identifiers or AMS should be there in channel connector.

File: src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java [hunk 8]
```diff
@@ -302,19 +318,33 @@ private void collectionRoutes(){
                         phoneNumber = ((JSONObject) payer.get(1)).getString("value");
                         accountId = ((JSONObject) payer.get(0)).getString("value");
                     }
+                    if(amsIdentifier.equals(amsUtils.ACCOUNTID)){
+                        //go for mpesa workflow
+                         tenantSpecificBpmn = mpesaFlow.replace("{dfspid}", tenantId).replace("{ams}",amsUtils.AMS_SECONDARY);
```
Comment (avikganguly01) [inline src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java:323]: AMS and Identifiers are hardcoded.

**Review by avikganguly01 — CHANGES_REQUESTED** [review]

File: src/main/java/org/mifos/connector/channel/camel/utils/AMSUtils.java (hunk 5 above)
Comment (ankita10r) [reply src/main/java/org/mifos/connector/channel/camel/utils/AMSUtils.java:42]: Yes, I missed this, will rename it.

**Review by avikganguly01 — APPROVED** [review]

**Pull request merged**